src/plugins/cache/cache.c

```diff
@@ -267,16 +289,29 @@ int elektraCacheGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key * pa
 		return ELEKTRA_PLUGIN_STATUS_SUCCESS;
 	}
 	// get all keys
-
 	CacheHandle * ch = elektraPluginGetData (handle);
 	if (ch->cacheStorage->global == 0)
 	{
 		ch->cacheStorage->global = elektraPluginGetGlobalKeySet (handle);
 	}
 
+	if (!elektraStrCmp (keyString (keyGetMeta (parentKey, "cacheClear")), "YES"))
+	{
+		// clear all caches
+		Key * cacheFile = keyDup (parentKey);
+		char * cacheFileName = kdbCacheFileName (ch, cacheFile, modeDirectory);
+		ELEKTRA_LOG_DEBUG ("CLEAR CACHES path: %s", cacheFileName);
+
+		keySetString (cacheFile, cacheFileName);
+		nftw (cacheFileName, unlinkCacheFiles, MAX_FD_USED, FTW_DEPTH);
```

Comment:
We should disable the plugin if nftw is not present see tests/cframework/CMakeLists.txt